feat(shared-data): add schema definitions for liquidProbeParameters

aaron-kulkarni opened the pull request:
The definitions of minimumHeight and minimumWellVolume are for all non tip rack labware. They will be used in LLD.

fix EXEC-518

<!--
Thanks for taking the time to open a pull request! Please make sure you've read the "Opening Pull Requests" section of our Contributing Guide:

https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

To ensure your code is reviewed quickly and thoroughly, please fill out the sections below to the best of your ability!
-->

# Overview

<!--
Use this section to describe your pull-request at a high level. If the PR addresses any open issues, please tag the issues here.
-->

# Test Plan

<!--
Use this section to describe the steps that you took to test your Pull Request.
If you did not perform any testing provide justification why.

OT-3 Developers: You should default to testing on actual physical hardware.
Once again, if you did not perform testing against hardware, justify why.

Note: It can be helpful to write a test plan before doing development

Example Test Plan (HTTP API Change)

- Verified that new optional argument `dance-party` causes the robot to flash its lights, move the pipettes,
then home.
- Verified that when you omit the `dance-party` option the robot homes normally
- Added protocol that uses `dance-party` argument to G-Code Testing Suite
- Ran protocol that did not use `dance-party` argument and everything was successful
- Added unit tests to validate that changes to pydantic model are correct

-->

# Changelog

<!--
List out the changes to the code in this PR. Please try your best to categorize your changes and describe what has changed and why.

Example changelog:
- Fixed app crash when trying to calibrate an illegal pipette
- Added state to API to track pipette usage
- Updated API docs to mention only two pipettes are supported

IMPORTANT: MAKE SURE ANY BREAKING CHANGES ARE PROPERLY COMMUNICATED
-->

# Review requests

<!--
Describe any requests for your reviewers here.
-->

# Risk assessment

<!--
Carefully go over your pull request and look at the other parts of the codebase it may affect. Look for the possibility, even if you think it's small, that your change may affect some other part of the system - for instance, changing return tip behavior in protocol may also change the behavior of labware calibration.

Identify the other parts of the system your codebase may affect, so that in addition to your own review and testing, other people who may not have the system internalized as much as you can focus their attention and testing there.
-->

aaron-kulkarni committed:
feat(shared-data): add schema definitions for minimumHeight and minimumWellVolume

The definitions of minimumHeight and minimumWellVolume are for all non tip rack labware. They will
be used in LLD.

fix EXEC-518

aaron-kulkarni committed:
deleted unneeded class

aaron-kulkarni committed:
add params to LabwareParameters

sfoster1 commented on shared-data/labware/schemas/2.json:
```diff
@@ -173,6 +173,14 @@
           "description": "Distance to move magnetic module magnets to engage",
           "$ref": "#/definitions/positiveNumber"
+        },
+        "minimumHeight:": {
```
this unfortunately needs to be more than just one number to handle labwares that have multiple different kinds of wells like the 15/50 combo tube rack. there's two options here:
1. It's part of the well datastructure, for each well
2. instead of containing a number, this is an array of objects that have a `value` member that is a number and an `appliesToWells` member that is an array of well names or something

Goes for both `minimumHeight` and `minimumWellVolume`.

sfoster1 reviewed (dismissed):
I think we need to change the structure a little to handle heterogenous-well labware like the 15/50 tuberack.

ryanthecoder commented:
I'd also like to see these be a joined object like
` "LiquidProbeSettings: {
            "minimumHeight": 0.5,
            "minimumWellVolume": 0,
  }`

aaron-kulkarni committed:
Restructured parameters into one object

aaron-kulkarni committed:
minor edit to types.ts

aaron-kulkarni committed:
minor formatting/refactoring changes

github-actions[bot] committed:
fix(app-testing): snapshot failure capture (#15462)

This PR is an automated snapshot update request. Please review the
changes and merge if they are acceptable or find you bug and fix it.

Co-authored-by: aaron-kulkarni <aaron-kulkarni@users.noreply.github.com>

aaron-kulkarni committed:
Merge branch 'edge' into exec-518-add-volume-labware-schema

aaron-kulkarni committed:
change default_factory of liquidProbeSettings

aaron-kulkarni committed:
more refactoring changes to parameters

aaron-kulkarni committed:
changed tuple to typeddict for parameters

github-actions[bot] committed:
fix(app-testing): snapshot failure capture (#15464)

This PR is an automated snapshot update request. Please review the
changes and merge if they are acceptable or find you bug and fix it.

Co-authored-by: aaron-kulkarni <aaron-kulkarni@users.noreply.github.com>

aaron-kulkarni committed:
minor formatting changes

aaron-kulkarni committed:
js lint

aaron-kulkarni committed:
actually fixing js lint

aaron-kulkarni committed:
update pd e2e tests

aaron-kulkarni committed:
final? lint change

ryanthecoder reviewed (approved):
Looks good to me
